blackjax/mcmc/random_walk.py

```diff
@@ -0,0 +1,296 @@
+
+    position: PyTree
+    logdensity: float
+
+
+class RWInfo(NamedTuple):
+    """Additional information on the RW chain.
+
+    This additional information can be used for debugging or computing
+    diagnostics.
+
+    acceptance_rate
+        The acceptance probability of the transition, linked to the energy
+        difference between the original and the proposed states.
+    is_accepted
+        Whether the proposed position was accepted or the original position
+        was returned.
+    proposal
+        The state proposed by the proposal.
+
+    """
+
+    acceptance_rate: float
+    is_accepted: bool
+    proposal: RWState
+
+
+def init(position: PyTree, logdensity_fn: Callable) -> RWState:
+    """Create a chain state from a position.
+
+    Parameters
+    ----------
+    position: PyTree
+        The initial position of the chain
+    logdensity_fn: Callable
+        Log-probability density function of the distribution we wish to sample
+        from.
+
+    """
+    return RWState(position, logdensity_fn(position))
+
+
+def additive_step(random_step: Callable):
+    """Build a Random Walk Rosenbluth-Metropolis-Hastings kernel
+
+    Returns
+    -------
+    A kernel that takes a rng_key and a Pytree that contains the current state
+    of the chain and that returns a new state of the chain along with
+    information about the transition.
+    """
+
+    def one_step(
+        rng_key: PRNGKey, state: RWState, logdensity_fn: Callable
+    ) -> Tuple[RWState, RWInfo]:
+        def proposal_generator(key_proposal, position):
+            move_proposal = random_step(key_proposal, position)
+            new_position = jax.tree_util.tree_map(jnp.add, position, move_proposal)
+            return new_position
+
+        kernel = rmh(logdensity_fn, proposal_generator)
+        return kernel(rng_key, state)
+
+    return one_step
+
+
+def irmh(proposal_distribution: Callable) -> Callable:
+    """
+    Build an Independent Random Walk Rosenbluth-Metropolis-Hastings kernel. This implies
+    that the proposal distribution does not depend on the particle being mutated :cite:p:`wang2022exact`.
+
+    Parameters
+    ----------
+    proposal_distribution
+        A function that, given a PRNGKey, is able to produce a sample in the same
+        domain of the target distribution.
+
+    Returns
+    -------
+    A kernel that takes a rng_key and a Pytree that contains the current state
+    of the chain and that returns a new state of the chain along with
+    information about the transition.
+
+    """
+
+    def one_step(
+        rng_key: PRNGKey, state: RWState, logdensity_fn: Callable
+    ) -> Tuple[RWState, RWInfo]:
+        def proposal_generator(rng_key: PRNGKey, position: PyTree):
+            return proposal_distribution(rng_key)
+
+        kernel = rmh(logdensity_fn, proposal_generator)
+        return kernel(rng_key, state)
+
+    return one_step
+
+
+def metropolis(logdensity_fn: Callable, transition_generator: Callable):
```

Comment:
I know per our conversations we wanted to make available a Metropolis only kernel. But looking at it here it seems unnecessary, good detailed documentation on `rmh` should give the user access to a Metropolis kernel and (unless you make tests for this kernel) it only reduces test coverage.